clang/lib/Driver/ToolChains/HIPAMD.cpp

```diff
@@ -209,6 +210,13 @@ void AMDGCN::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   if (JA.getType() == types::TY_LLVM_BC)
     return constructLlvmLinkCommand(C, JA, Inputs, Output, Args);
 
+  if (Args.getLastArgValue(options::OPT_mcpu_EQ) == "generic") {
+    llvm::opt::ArgStringList TrArgs{"--spirv-max-version=1.1",
```

Comment:
I'm not sure we want to stick with 1.1 here, the Translator goes up to 1.4 at the moment - should we consider going to that instead?

Reply:
I wonder if `-mcpu` is the correct way to encode this. Targeting SPIR-V is more like the triple than the architecture as far as I'm aware.

Reply:
> I'm not sure we want to stick with 1.1 here, the Translator goes up to 1.4 at the moment - should we consider going to that instead?

Thanks for reminder. I think we should be able to go up with the version since we will use ToT of the LLVM/SPIRV translator.

Reply:
> I wonder if `-mcpu` is the correct way to encode this. Targeting SPIR-V is more like the triple than the architecture as far as I'm aware.

I will see whether I can use triple instead.